network/gossip/libp2p/sporking_test.go

```diff
@@ -118,18 +108,129 @@ func (h *SporkingTestSuite) TestProtocolChangeAfterHardSpoon() {
 	// start node2 with the same name, ip and port but with the new key
 	node2, address2New := h.CreateNode(node2.name, node2key, address2.IP, address2.Port, rootID2, nil, false)
 	defer h.StopNode(node2)
-	h.T().Logf(" %s node again started on %s:%s", node2.name, address2New.IP, address2New.Port)
-	h.T().Logf("new libp2p ID for %s: %s", node2.name, node2.libP2PHost.ID())
 
 	// make sure the node2 indeed came up on the old ip and port
 	assert.Equal(h.T(), address2.IP, address2New.IP)
 	assert.Equal(h.T(), address2.Port, address2New.Port)
 
 	// attempt to create a stream from node 2 (new chain) to node 1 (old chain)
 	// this time it should fail since node 2 is listening on a different protocol
-	_, err = node2.CreateStream(context.Background(), address1)
+	testOneToOneMessagingFails(h.T(), node2, address1)
+}
+
+// TestOneToKCrosstalkPrevention tests that a node from the old chain cannot talk to a node in the new chain via PubSub
+// if the channel ID is updated while the network keys are kept the same.
+func (h *SporkingTestSuite) TestOneToKCrosstalkPrevention() {
+
+	// root id before spork
+	rootIDBeforeSpork := unittest.BlockFixture().ID().String()
+
+	// create and start node 1 on localhost and random port
+	node1key, err := generateNetworkingKey("abc")
+	assert.NoError(h.T(), err)
+	node1, _ := h.CreateNode("node1", node1key, "0.0.0.0", "0", rootIDBeforeSpork, nil, false)
+	defer h.StopNode(node1)
+
+	// create and start node 2 on localhost and random port with the same root block ID
+	node2key, err := generateNetworkingKey("def")
+	assert.NoError(h.T(), err)
```

Comment:
suggest `require.NoError` instead, also other places in the code.

Reply:
changed all assert.NoError to require.NoError